packages/expo-image/ios/ResizedAnimatedCoder.swift

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024-present 650 Industries. All rights reserved.
+
+import SDWebImage
+
+/**
+ An animated image resizer that adjusts an already loaded `SDAnimatedImage`.
+ This is not a formal SDWebImage coder; it's used solely to leverage the SDWebImage loading pipeline.
+ */
+internal final class ResizedAnimatedCoder: SDImageIOAnimatedCoder {
+  private let image: SDAnimatedImage
```

Comment:
The coder should not retain the `SDAnimatedImage`. Because this cause retain cycle.

SDAnimatedImage will retain the `animatedCoder` you passed into the initializer.

Reply:
as mentioned above, we didn't go through decode in this coder. the `initWithAnimatedImageData:options:` has not being called, so data from `super` is not available.

current approach doesn't have retain cycle. it's like
`resized SDWebImage` -> `ResizedAnimatedCoder` -> `ResizedAnimatedCoder.image (original image)` -> `original animatedCoder (for example, SDImageAVIFCoder)`
if not don't retain the image here, the original image will be deallocated.